hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/AbstractInternalQueryCache.java

```diff
@@ -164,7 +164,7 @@ protected void doFullEntryScan(Predicate predicate, Set<Map.Entry<K, V>> resulti
             boolean valid = predicate.apply(queryEntry);
             if (valid) {
                 Object keyObject = queryEntry.getKey();
-                Object valueObject = queryEntry.getValue();
+                Object valueObject = toObject(queryEntry.getValueData());
                 Map.Entry simpleEntry = new AbstractMap.SimpleEntry(keyObject, valueObject);
```

Comment:
maybe we can use `LazyMapEntry` as in [ClientMapProxy#entrySet](https://github.com/hazelcast/hazelcast/blob/6a5bc11894e312366e82d4c808df31c2d441d0fc/hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java#L1314)

Reply:
done